store/postgres/src/chain_store.rs

```diff
@@ -1813,6 +1841,120 @@ impl ChainStoreTrait for ChainStore {
     }
 }
 
+mod recent_blocks_cache {
+    use super::*;
+
+    type OffsetFromChainHead = BlockNumber;
+
+    struct Inner {
+        blocks: HashMap<OffsetFromChainHead, (BlockPtr, Option<json::Value>)>,
+        capacity: usize,
+    }
+
+    /// We cache the most recent blocks in memory to avoid overloading the
+    /// database with unnecessary queries close to the chain head. We invalidate
+    /// blocks:
+    /// - After 5 seconds.
+    /// - Whenever the chain head advances.
```

Comment:
That comment needs to be updated since there is no TTL anymore

Reply:
Done, good catch!